rust/crd/src/lib.rs

```diff
@@ -411,6 +413,10 @@ impl SparkApplication {
             .as_ref()
             .and_then(|executor_config| executor_config.node_selector.clone())
     }
+
+    pub fn job_node_selector(&self) -> Option<std::collections::BTreeMap<String, String>> {
+        self.spec.node_selector.clone()
+    }
```

Comment:
Have you considered deriving this from the driver's node_selector instead of making it configurable ? The fact that we use a Job to manage the Spark application is an implementation detail and we we should try not to leak it. For the user is not clear what this nodeSelector is actually for.

Reply:
Good point.

Reply:
Suggestion implemented.